### DISCUSSION TITLE
[4.0] [Feature Request] Version storage for the HTML module

### DISCUSSION THREAD

**Discussion by Gnevyshev** [joomla/joomla-cms#28686] [2. Cold Features]
Hello!
By analogy with the storage of content (articles) versions, I propose to provide for the storage of versions for HTML modules.
It would be very useful for me! I think, not only for me.

**Comment by richard67** [comment]
As Joomla 3 is locked for new features, it will go into Joomla 4 if it will be implemented. I will add labels and so on that it's clear that it's a feature request and that it's for Joomla 4.

**Comment by brianteeman** [comment]
I am sure this has been discussed before

**Comment by richard67** [comment]
@brianteeman Seems so: #11136 ... and this points to [https://github.com/joomla/joomla-cms/issues/7019#issuecomment-217729101](https://github.com/joomla/joomla-cms/issues/7019#issuecomment-217729101). Is that still valid?

From my point of view it would **not** be a small task to integrate modules into content history, and making an own kind of history would be even bigger, so maybe the answer will stay the same as it was.

**Comment by brianteeman** [comment]
maybe not as I "think" we've broken the connection between ucm and versions now?

**Comment by richard67** [comment]
Hmm, then I am not up to date it seems.

**Comment by richard67** [comment]
Who could know more about that?

**Comment by richard67** [comment]
Not that I am against it ... I think it would be really useful.

**Comment by brianteeman** [comment]
I was getting confused with a proposal for tags https://github.com/joomla/joomla-cms/pull/24551

In which case the original comments are probably still valid.

**Comment by richard67** [comment]
Possibly you got confused because it is the same expert for both ;-)

@Hackwar What do you think about content history for (custom html) modules (or their content)? Would that be implementable with a reasonable effort on brain cells, coffee and work time?

**Comment by Hackwar** [comment]
We could look at this in 4.1+. For 4.0 this comes to late.

**Comment by Gnevyshev** [comment]
Thank you for interest! I will hope.
